etc/lfcd.fish

```diff
@@ -11,17 +11,7 @@
 #
 # You may put this in a function called fish_user_key_bindings.
 
-function lfcd
-    set tmp (mktemp)
+function lfcd --wraps="lf" --description="lf - Terminal file manager (changing directory on exit)"
     # `command` is needed in case `lfcd` is aliased to `lf`
-    command lf -last-dir-path=$tmp $argv
-    if test -f "$tmp"
-        set dir (cat $tmp)
-        rm -f $tmp
-        if test -d "$dir"
-            if test "$dir" != (pwd)
-                cd $dir
-            end
-        end
-    end
+    cd (command lf -print-last-dir $argv)
```

Comment:
What would happen if lf didn't print anything to stdout, e.g. crashed? I wouldn't want to cd to the home dir in that case.

Other than that, this lgtm; I think getting an error from `cd` when the dir is not valid is fine.

Reply:
Updated it 👍

Reply:
Although somethine like `lfcd --doc` produces errors, do you think it falls under "I think getting an error from cd when the dir is not valid is fine"?

Reply:
The updated code looks better to me. Apart from not cd-ing on failure, I'm more interested in that the following does not `cd` you to the home dir:

```fish
🐟 set -g outdir ""
🐟 cd $outdir
cd: Empty directory '' does not exist
```

Alternatively, you could insert a check that $outdir is not empty. But, at that point, you could just again check whether the dir exists. I think I'm OK either way.

Thank you!

------

As an aside, I checked that both of the following `cd` you to the home dir, so the fix is useful.

```fish
cd (printf "")
cd (printf ""; false)
```

Reply:
@ilyagr Good catch about `cd`-ing back to the home directory in case `lf` crashes, I didn't consider that. Just out of curiosity, would it not be better to quote the command substitution (i.e. `"$(...)"`), to ensure the output is a blank string instead of an empty list?

I found the following commands result in a `cd: Empty directory '' does not exist` error:

```fish
cd "$(printf '')"
cd "$(false)"
```

Reply:
@ilyagr sorry, I don't follow: are you suggesting to introduce a check for whether the resulting dir variable is empty?

Reply:
@postsolar I think the point @ilyagr is trying to make is that under no circumstances should `lfcd` change to the home directory in the case that `lf` crashes  or otherwise prints nothing. I think a local variable is not needed and the below should work:

```fish
cd "$(command lf -print-last-dir $argv)"
```

Reply:
@postsolar, sorry, I missed your question. I think @joelim-work covered it, thanks Joe!

Reply:
@joelim-work @ilyagr thank you for clarifying! I updated the code and added a comment